### PR TITLE
 Dreamview: added drive event types

### PR DESCRIPTION
**Pull request by vlin17**

### CODE REVIEW

File: modules/dreamview/backend/hmi/hmi_worker.cc [hunk 1]
```diff
@@ -203,11 +204,17 @@ int HMIWorker::RunToolCommand(const std::string &tool,
...
 void HMIWorker::SubmitDriveEvent(const uint64_t event_time_ms,
-                                 const std::string &event_msg) {
-  apollo::common::DriveEvent drive_event;
+                                 const std::string &event_msg,
+                                 const std::vector<std::string> &event_types) {
+  DriveEvent drive_event;
   AdapterManager::FillDriveEventHeader("HMI", &drive_event);
   drive_event.mutable_header()->set_timestamp_sec(event_time_ms / 1000.0);
   drive_event.set_event(event_msg);
+  for (const auto type_name : event_types) {
```
Comment (siyangy) [inline modules/dreamview/backend/hmi/hmi_worker.cc]: const auto&

File: modules/dreamview/backend/hmi/hmi_worker.cc [hunk 2]
```diff
@@ -203,11 +204,17 @@ int HMIWorker::RunToolCommand(const std::string &tool,
...
+    DriveEvent::Type type;
+    DriveEvent::Type_Parse(type_name, &type);
```
Comment (siyangy) [inline modules/dreamview/backend/hmi/hmi_worker.cc]: What if this parse fails? Will it crash the server?

Comment (vlin17) [reply modules/dreamview/backend/hmi/hmi_worker.cc]: added checker

File: modules/dreamview/backend/hmi/hmi_worker.cc (hunk 1 above)
Comment (vlin17) [reply modules/dreamview/backend/hmi/hmi_worker.cc]: done.

**Pull request merged**